### PR TITLE
fix the mypy plugin assertion error when enums, unions, etc are not used as decorator

### PR DESCRIPTION
**Pull request by andriykohut**
## Description

When `strawberry.enum` or `strawberry.union` is not used as decorator, mypy fails with a following traceback:
```
Traceback (most recent call last):
  File "mypy/semanal.py", line 5132, in accept
  File "mypy/nodes.py", line 1140, in accept
  File "mypy/semanal.py", line 2110, in visit_assignment_stmt
  File "mypy/semanal.py", line 2392, in apply_dynamic_class_hook
  File "/Users/andrii.kohut/code/strawberry/strawberry/ext/mypy_plugin.py", line 229, in enum_hook
    enum_type = _get_type_for_expr(first_argument, ctx.api)
  File "/Users/andrii.kohut/code/strawberry/strawberry/ext/mypy_plugin.py", line 115, in _get_type_for_expr
    return _get_named_type(expr.name, api)
  File "/Users/andrii.kohut/code/strawberry/strawberry/ext/mypy_plugin.py", line 99, in _get_named_type
    return api.named_type(name)
  File "mypy/semanal.py", line 4538, in named_type
  File "mypy/semanal.py", line 4504, in lookup_fully_qualified
  File "mypy/semanal.py", line 4520, in lookup_fully_qualified_or_none
AssertionError:
```

Examples:
```py
import enum

import strawberry

class TestEnum(enum.Enum):
    a = enum.auto()
    b = enum.auto()
    c = enum.auto()

EnumType = strawberry.enum(TestEnum)
```
or
```py
import strawberry

@strawberry.type
class A:
    x: int

@strawberry.type
class B:
    x: int

Test = strawberry.union("Test", (A, B))
```

Assertion that breaks the plugin: https://github.com/python/mypy/blob/9d5ef72068943e415082c848b74ff4fad9293080/mypy/semanal.py#L4508-L4520

## Types of Changes

- [ ] Core
- [x] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [ ] Documentation

## Issues Fixed or Closed by This PR

*

## Checklist

- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have read the CONTRIBUTING document.
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

**Comment by botberry** [comment]
Hi, thanks for contributing to Strawberry 🍓!

We noticed that this PR is missing a `RELEASE.md` file. We use that to automatically do releases here on GitHub and, most importantly, to PyPI!

So as soon as this PR is merged, a release will be made 🚀.

Here's an example of `RELEASE.md`:

```markdown
Release type: patch

Description of the changes, ideally with some examples, if adding a new feature.
```

Release type can be one of patch, minor or major. We use [semver](https://semver.org/), so make sure to pick the appropriate type. If in doubt feel free to ask :)


Here's the tweet text: 
```
🆕 Release (next) is out! Thanks to Andrii Kohut for the PR 👏

Get it here 👉 https://github.com/strawberry-graphql/strawberry/releases/tag/(next)
```

**Review by patrick91 — COMMENTED** [review]
Oh awesome! Would you be able to add a test here? https://github.com/strawberry-graphql/strawberry/tree/main/tests/mypy

**Comment by andriykohut** [comment]
@patrick91 Weirdly it seems there are some similar cases, but they weren't falling for some reason, i.e. https://github.com/strawberry-graphql/strawberry/blob/43e4343687ca442a35ab3ab473675e337df98ee0/tests/mypy/test_enum.yml#L12

I guess it's because it's executed via mypy pytest plugin, which always adds `main.IceCreamFlavour`, not sure how to test it properly via the plugin 🤔

File: RELEASE.md [hunk 1]
```diff
@@ -0,0 +1,3 @@
+Release type: bugfix
```
Comment (patrick91) [inline RELEASE.md:1]:
```suggestion
Release type: patch
```

**Review by patrick91 — APPROVED** [review]

**Comment by patrick91** [comment]
> @patrick91 Weirdly it seems there are some similar cases, but they weren't falling for some reason, i.e.
> 
> https://github.com/strawberry-graphql/strawberry/blob/43e4343687ca442a35ab3ab473675e337df98ee0/tests/mypy/test_enum.yml#L12
> 
> I guess it's because it's executed via mypy pytest plugin, which always adds `main.IceCreamFlavour`, not sure how to test it properly via the plugin 🤔

ok! I guess it is fine, maybe we'll find a different way to run mypy tests in future 😊

File: strawberry/ext/mypy_plugin.py [hunk 2]
```diff
@@ -112,7 +112,7 @@ def _get_type_for_expr(expr: Expression, api: SemanticAnalyzerPluginInterface) -
             if sym and isinstance(sym.node, Var):
                 raise InvalidNodeTypeException(sym.node)
 
-        return _get_named_type(expr.name, api)
+        return _get_named_type(expr.fullname or expr.name, api)
```
Comment (patrick91) [inline strawberry/ext/mypy_plugin.py:115]: 👍

**Comment by botberry** [comment]
Thanks for contributing to Strawberry! 🎉 You've been invited to join
the Strawberry GraphQL organisation 😊

You can also request a free sticker by filling this form: https://forms.gle/dmnfQUPoY5gZbVT67

And don't forget to join our discord server: https://strawberry.rocks/discord 🔥

**Pull request merged**